spv/rescan.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2013-2014 The btcsuite developers
+// Copyright (c) 2018-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package spv
+
+import (
+	"github.com/decred/dcrd/blockchain/stake/v2"
+	"github.com/decred/dcrd/gcs/blockcf"
+	"github.com/decred/dcrd/txscript/v2"
+	"github.com/decred/dcrd/wire"
+)
+
+// rescanCheckTransaction is a helper function to rescan both stake and regular
+// transactions in a block.  It appends transactions that match the filters to
+// *matches, while updating the filters to add outpoints for new UTXOs
+// controlled by this wallet.  New data added to the Syncer's filters is also
+// added to fadded.
+//
+// This function may only be called with the filter mutex held.
+func (s *Syncer) rescanCheckTransactions(matches *[]*wire.MsgTx, fadded *blockcf.Entries, txs []*wire.MsgTx, tree int8, walletID int) {
```

Comment:
The file referenced was part of the code gotten from dcrwallet's SPV module and I'm avoiding making a lot of changes to it so that there are little conflicts when updating from upstream.